llvm/lib/Target/AIE/AIEAlternateDescriptor.h

```diff
@@ -0,0 +1,74 @@
+//== AIEAlternateDescriptor.h - Define Alternate descriptor Class *-C++-*-===//
+//
+// This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// (c) Copyright 2024 Advanced Micro Devices, Inc. or its affiliates
+//
+//===----------------------------------------------------------------------===//
+//
+// This file declares the AIEngine Alternate instruction descriptor class
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef LLVM_SUPPORT_AIEALTERNATEDESCRIPTOR_H
+#define LLVM_SUPPORT_AIEALTERNATEDESCRIPTOR_H
+
+#include "AIEBaseSubtarget.h"
+#include "MCTargetDesc/AIEMCFormats.h"
+
+#include <unordered_map>
+
+namespace llvm {
+
+using MIAltDescsMap = std::unordered_map<MachineInstr *, const MCInstrDesc *>;
+
+class AIEAlternateDescriptor {
```

Comment:
I think the name makes it sound like this data structure stores a single descriptor. Maybe rename to `AIEAlternateDescriptors` or `AIEAlternateDescriptorMap`?